File: services/gateway-service/src/main/java/com/atlanta/banking/gateway/service/ErrorResponse.java
package com.atlanta.banking.gateway.service;

import lombok.*;

import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.Map;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ErrorResponse {

    private LocalDateTime timestamp;

    private int status;

    private String error;

    private String message;

    private String path;

    @Builder.Default
    private Map<String, String> fieldErrors = new HashMap<>();
}

File: notification-service/app/schemas/error_response.py
from datetime import UTC, datetime

from pydantic import BaseModel


class ValidationErrorDetail(BaseModel):
    field: str
    message: str


class ErrorResponse(BaseModel):
    timestamp: datetime = datetime.now(UTC)
    status: int
    error: str
    message: str
    path: str
    errors: list[ValidationErrorDetail] | None = None